addons/html_editor/static/src/main/media/media_plugin.js

```diff
@@ -341,4 +344,18 @@ export class MediaPlugin extends Plugin {
             el.setAttribute("src", newAttachmentSrc);
         }
     }
+    markMediaAsEditable(media) {
+        media.classList.add(EDITABLE_MEDIA_CLASS);
+    }
+    isMediaEditable(media) {
+        return media.classList.contains(EDITABLE_MEDIA_CLASS);
+    }
+    isNodeEditableMedia(node) {
+        return (isMediaElement(node) || node.nodeName === "IMG") && this.isMediaEditable(node);
+    }
+    isNodeEditable(node) {
+        if (isMediaElement(node) || node.nodeName === "IMG") {
+            return this.isMediaEditable(node);
+        }
+    }
```

Comment:
Now that I think about it, why didn't we simply put `contenteditable="true"` on the image itself? It's not like we can write inside the image anyway and, conceptually, it could be seen as editing the "content" of the html image, that is the editing the actual image file it points to.

Also, none of these functions are used, but I suppose they will be in later commits.

Reply:
For the image to be selected we (currently and in my branch) need the selection to be `<figure>[<img>]<figcaption/></figure>` which is `(figure, 0, figure, 1)` so we'd the figure to be contenteditable, not the image. `(img, 0, img, 0)` wouldn't mean much and would seem collapsed. Our system currently checks whether the node we're referencing in the selection (`figure` in the case of (figure, 0, figure 1)) is contenteditable.
Or did you mean more symbolically, to use the contenteditable attribute instead of a class and keep the rest of my implementation the same?

They will indeed be used in later commits, when the caption feature gets introduced.

Reply:
Yes I meant symbolically.

Reply:
I'm not against it on principle but wouldn't that be even more confusing as it's attached to a separate concept?

Reply:
Also I'd be weary of side effects, be it from Phoenix or from the browser. You never know ^^'